Bugfix/cam rev10

MarcelCaron-NOAA opened the pull request:
## Pull Request Testing ##

- [x] Describe testing already performed for this Pull Request:</br>

This PR addresses issue 86.35-36 noted in the Google Doc titled "EVS v1.0 issues".  Information about its cause and fix:

* 86.35 - These jobs output missing file messages when some files are not gathered.  This is usually OK, but a "WARNING" still needs to be printed, and was not. _Fix_: Add "WARNING" to the message, along with some context.
* 86.36 - Two missing file messages hidden in SENDMAIL if blocks were missed in PR445.  _Fix:_ Add a WARNING message outside each if block.

The affected jobs were not tested (but run quickly).

---
- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>



For other jobs:
### (1) Set up jobs 
•	symlink the EVS_fix directory locally as "fix"
•	In all driver scripts, point all exports of "HOMEevs" to your EVS test directory
•	In all driver scripts, change COMIN to the parallel directory (`/lfs/h2/emc/vpppg/noscrub/emc.vpppg/$NET/$evs_ver_2d`)
•	In all driver scripts, set SENDMAIL to "NO"
•	In all driver scripts, set envir to "pppp"
•	In all driver scripts, add the following line: `export DCOMROOT=/lfs/h1/ops/pppp/dcom`

### (2) Test Jobs
Use the following to submit all jobs: `qsub –v vhr=00 `

From your EVS testing directory, the following scripts should be run using the setup in step (1):
```
dev/drivers/scripts/prep/cam/jevs_cam_hireswarw_precip_prep.sh
dev/drivers/scripts/prep/cam/jevs_cam_severe_prep.sh
```

_[Total: 2 prep jobs]_


### (3) Review

jevs_cam_hireswarw_precip_prep:

- `grep "Input.*file does not exist"`
- look for "WARNING" attached to this message

jevs_cam_severe_prep:
- `grep "'[' -s .*spc_reports"\
- look for a "WARNING" message below this log statement

Both jobs:
- check for FATAL ERROR: `grep -l "FATAL ERROR" $outfile`
- check for no WARNINGs: `grep -L "WARNING" $outfile`

Neither job should output any FATAL ERRORs.  Both jobs should output WARNINGs indicating missing files.

Nominal runs can use the usual check:
```
check="FATAL\|WARNING\|error\|Killed\|Cgroup\|argument expected\|No such file\|cannot\|failed\|unexpected\|exceeded"
grep "$check" $outfile
```

---
- [x] Has the code been checked to ensure that no errors occur during the execution? **YES**

---
- [x] Do these updates/additions include sufficient testing updates? **YES**

---
- [x] Please complete this pull request review by **02/14/2024**.</br>

---
## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

Commit:
Fix item 86.36
Add WARNING log outside of mail message if block

Commit:
Fix item 86.35
Add "WARNING" and reassuring message

Commit:
Fix item 86.35
Add "WARNING" to message about missing ccpa data

ShelleyMelchior-NOAA commented on scripts/prep/cam/exevs_cam_severe_prep.sh:
```diff
@@ -54,6 +54,7 @@ fi
 # Send missing data alert if needed
 if [ $data_missing ]; then
 
+   echo "WARNING: File $DCOMINspc/${OTLK_DATE}/validation_data/weather/spc/day*otlk_{OTLK_DATE}*.zip is missing" >> mailmsg
```
I think the redirect of this echo statement to mailmsg should be removed.

MarcelCaron-NOAA replied:
@ShelleyMelchior-NOAA Thanks for catching this.  Removed!

PerryShafran-NOAA commented:
@MarcelCaron-NOAA I made the change that Shelley suggested locally and am running these two tests.  You will still need to make that change into your fork.  

Also you will need to sync your fork.

PerryShafran-NOAA commented:
@MarcelCaron-NOAA I think that changing envir=pppp doesn't do much for the spc_reports, since that points to dcom, not com, and thus doesn't use envir in its directory.  See the file /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr448test/EVS/dev/drivers/scripts/prep/cam/jevs_cam_severe_prep.o132710111

```
49 + '[' -s /lfs/h1/ops/prod/dcom/20240205/validation_data/weather/spc/spc_reports_20240205.csv ']'
49 + mkdir -p /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/prep/cam/spc_lsr.20240205
```

Also we get WARNINGs, but they are METplus warnings.  

The hiresw precip prep .o file looks like it has what you needed.

Commit:
Fix 86.46
Remove mailmsg redirect

ShelleyMelchior-NOAA commented:
> @MarcelCaron-NOAA I think that changing envir=pppp doesn't do much for the spc_reports, since that points to dcom, not com, and thus doesn't use envir in its directory. See the file /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr448test/EVS/dev/drivers/scripts/prep/cam/jevs_cam_severe_prep.o132710111
> 
> ```
> 49 + '[' -s /lfs/h1/ops/prod/dcom/20240205/validation_data/weather/spc/spc_reports_20240205.csv ']'
> 49 + mkdir -p /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/prep/cam/spc_lsr.20240205
> ```
> 
> Also we get WARNINGs, but they are METplus warnings.
> 
> The hiresw precip prep .o file looks like it has what you needed.

@PerryShafran-NOAA , you can always just export DCOMIN directly from the driver to something phony if there is any wonkiness w/ $envir. But also, DCOMIN does indeed have prod ($envir) in its path ...

Commit:
Merge branch 'develop' into bugfix/cam_rev10

AliciaBentley-NOAA commented:
@MarcelCaron-NOAA Please check the EVS v1.0 Issues doc for the comment where you said you could add a WARNING in the same way you did for another part of EVS. Wei would like you to add it. Thanks!

@ShelleyMelchior-NOAA @PerryShafran-NOAA

MarcelCaron-NOAA commented:
@AliciaBentley-NOAA Okay, thanks.  Yes I figured Wei would say yes and I added it in already.  I can confirm the WARNING triggered in the jevs_cam_hireswarw_precip_prep test job here.

MarcelCaron-NOAA commented:
@PerryShafran-NOAA Yes it looks like DCOMROOT needs to be set to something like 

```
export DCOMROOT=/lfs/h1/ops/pppp/dcom
```
for these tests to work, since the spc files use DCOMIN.  

I haven't been testing this way myself.  I just did now, inserting that line in the driver script, and it seems to work!

ShelleyMelchior-NOAA commented:
> @PerryShafran-NOAA Yes it looks like DCOMROOT needs to be set to something like
> 
> ```
> export DCOMROOT=/lfs/h1/ops/pppp/dcom
> ```
> 
> for these tests to work, since the spc files use DCOMIN.
> 
> I haven't been testing this way myself. I just did now, inserting that line in the driver script, and it seems to work!

I have been testing the way suggested just now. Just force a phony DCOMIN* for all variations of DCOMIN and force a phony COMIN* for all variations of COMIN. It may be a tiny bit  more work adding lines to the driver, but it doesn't introduce any nuance that may come up w/ altering $envir, which can be woven into other things, not just input data paths.

PerryShafran-NOAA commented:
@MarcelCaron-NOAA OK, check out jevs_cam_severe_prep.o132721229 to see if this checks out.

You didn't comment on the hireswarw precip prep script yet:  jevs_cam_hireswarw_precip_prep.o132710106

MarcelCaron-NOAA commented:
@PerryShafran-NOAA can we try one more test of jevs_cam_hireswarw_precip_prep with the bogus DCOMIN setting you used for the severe job?  I want to confirm the mrms warning message that Wei flagged.

Other than that:
✔️ jevs_cam_hireswarw_precip_prep shows WARNINGs for missing ccpa files
✔️ jevs_cam_severe_prep shows WARNINGs for spc_lsr and spc_otlk files

Neither have FATAL ERRORs.

PerryShafran-NOAA commented:
@MarcelCaron-NOAA Running with the DCOMROOT set to /lfs/h1/ops/pppp/dcom.

PerryShafran-NOAA commented:
@MarcelCaron-NOAA Please check the file jevs_cam_hireswarw_precip_prep.o132722178.  It is smaller than the other file.

MarcelCaron-NOAA commented:
@PerryShafran-NOAA FATAL ERROR:  I think because "vhr=00" wasn't fed into that job.  Can you run one more time?

PerryShafran-NOAA commented:
@MarcelCaron-NOAA Fixed and restarted.

MarcelCaron-NOAA commented:
✔️ jevs_cam_hireswarw_precip_prep shows WARNINGs for missing mrms files

Both data denial jobs addressed issues 86.1, 86.35, and 86.36.  Both complete with WARNINGs and no FATAL ERRORs.  No further concerns from me.

PerryShafran-NOAA commented:
@MarcelCaron-NOAA Check to see if you need a sync fork.

Commit:
Merge branch 'develop' into bugfix/cam_rev10

MarcelCaron-NOAA commented:
@PerryShafran-NOAA Synced 👍

PerryShafran-NOAA reviewed (approved):
Code works as expected.  Approved for merge.
